File: CodeChef/June17/UNIONSET.py
t = int(raw_input())

for i in xrange(t):
    n,k = map(int, raw_input().split())
    presence = []
    lengths = []
    for j in xrange(n):
        l = map(int, raw_input().split())
        lengths.append(l[0])
        temp = [False for _ in xrange(k+1)]
        for x in l[1:]:
            temp[x] = True
        presence.append(temp)
    ans = 0
    for first in xrange(n):
        for second in xrange(first+1, n):
            if lengths[first] + lengths[second] >= k:
                possible = True
                pf = presence[first]
                ps = presence[second]
                for x in xrange(1,k+1):
                    if not pf[x] and not ps[x]:
                        possible = False
                        break

                if possible:
                    ans += 1

    print ans

File: CodeChef/June17/unionset.c
#include <stdio.h>
int presence[2500][2501];
int main(void) {
    int t;
    scanf("%d", &t);
    int lengths[2500];
    for (int i = 0; i < t; i++) {
        int n,k;
        scanf("%d %d", &n, &k);
        for (int j = 0; j < n; j++) {
            for (int x = 0; x <= k; x++)
                presence [j][x] = 0;
        }
        for (int j = 0; j < n; j++) {
            int count;
            scanf("%d", &count);
            lengths[j] = count;
            for (int x = 0; x < count; x++) {
                int num;
                scanf("%d", &num);
                presence[j][num] = 1;
            }
        }
        int ans = 0;
        for (int first = 0; first < n; first++) {
            for (int second = first + 1; second < n; second++) {
                if (lengths[first] + lengths[second] >= k) {
                    int possible = 1;
                    for (int x = 1; x <= k ;x++) {
                        if (presence[first][x] == 0 && presence[second][x] == 0) {
                            possible = 0;
                            break;
                        }
                    }
                    if (possible)
                        ans++;
                }
            }
        }

        printf("%d\n", ans);
    }
}